skbio/diversity/_fast_base.py

```diff
@@ -0,0 +1,54 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from skbio.diversity._fast_base_cy import nodes_by_counts
+
+
+def index_tree(tree):
+    """Index a tree to allow for bulk numpy aggregations
+
+    Paramters
+    ---------
+    tree : TreeNode
+        A tree to index
+
+    Returns
+    -------
+        dict of array
+            {id_index: {id: TreeNode},
+             child_index: [(node_id, left_child_id, right_child_id)],
+             attr_1: array(...),
+             ...
+             attr_N: array(...)}
+
+    Notes
+    -----
+    This wraps `TreeNode.to_array`, but replaces any `nan` in the length array
```

Comment:
Would it make more sense for this replacement to optionally occur as part of ``TreeNode.to_array``, rather than have a separate function for it here?